src/main/java/pl/allegro/tech/discovery/consul/recipes/watch/catalog/ServiceInstance.java

```diff
@@ -24,12 +26,12 @@ public List<String> getServiceTags() {
         return serviceTags;
     }
 
-    public String getServiceAddress() {
-        return serviceAddress;
+    public Optional<String> getServiceAddress() {
+        return Optional.ofNullable(serviceAddress);
```

Comment:
I guess that hashCode/equals do not need to be updated to use `.get()`?

Reply:
indeed not need